phases/05-nlp-foundations-to-advanced/10-attention-mechanism/code/main.py

```
@@ -0,0 +1,76 @@
+import math
+
+
+def softmax(scores):
+    m = max(scores)
+    exps = [math.exp(s - m) for s in scores]
+    total = sum(exps)
+    return [e / total for e in exps]
+
+
+def dot(a, b):
+    return sum(x * y for x, y in zip(a, b))
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)